sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/NettyAsyncHttpResponse.java

```diff
@@ -58,6 +64,34 @@ public Mono<InputStream> getBodyAsInputStream() {
         return bodyIntern().aggregate().asInputStream();
     }
 
+    @Override
+    public Mono<Void> transferBodyToAsync(AsynchronousByteChannel channel) {
+        return bodyIntern().retain()
+            .flatMapSequential(nettyBuffer ->
+                FluxUtil.writeToAsynchronousByteChannel(Mono.just(nettyBuffer.nioBuffer()).flux(), channel)
+                    .doFinally(ignored -> nettyBuffer.release()), 1, 1)
```

Comment:
With retaining and releasing manually do we need to continue deep buffering?

Reply:
It's by scenario.
In scenarios where we have full control over buffer lifecycle (like this one) we can avoid buffering. (which then maps to all the downloadTo(outputstream/file/channel) apis on the clients).
On in the future if we can build smarter input stream/byte channel on top of response stream.

The `Flux<ByteBuffer> HttpResponse.getBody` has to buffer as we give up control there.